Bottom: ab81e1f978e3ae55b11247f8b08f0d33ba076340
Top:    2fcffc01649b621e2be6a205293418c69850ada5
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:57:51 +0000

polygon1.c: Refactor poly_Copy0 to use poly_Create(), rather than open-coding

This ensures all polygon initialisation happens in a common place


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 5981433..1fbcabf 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2938,12 +2938,10 @@ poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
   if (src == NULL)
     return TRUE;
 
-  *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
